src/server/routes/entity/publisher.js

```diff
@@ -58,6 +58,17 @@ function _setPublisherTitle(res) {
 	);
 }
 
+/**
+ * Returns positive negative or zero value according to a and b.
+ * @param {object} a The first edition to compare.
+ * @param {object} b The second edition to compare.
+ * @returns {number} .
+ */
+function sortByDate(a, b) {
+	return new Date(a.releaseEventSet.releaseEvents[0].date).getTime() -
```

Comment:
You need to check whether a.releaseEventSet and b.releaseEventSet are null, and then whether releaseEvents has any elements. I'd suggest using:

    function compateEntitiesByDate(a, b) {
        const aDate = _.get(a, 'releaseEventSet.releaseEvents[0].date', null);
        const bDate = _.get(b, 'releaseEventSet.releaseEvents[0].date', null);
        if (_.isNull(aDate)) {
            // return a positive value, so that non-null dates always come before null dates.
            return 1; 
        }

        if (_.isNull(bDate)) {
            // return a negative value, so that non-null dates always come before null dates.
            return -1; 
        }

        return new Date(aDate).getTime() - new Date(bDate).getTime();
    }

Also, please place this in a common file (maybe `entity/entity.js`), and then import it to `entity/publisher.js` and `entity/publication.js`.